Sources/PromotedCore/ClientConfig.swift

```diff
@@ -103,6 +103,16 @@ public final class ClientConfig: NSObject {
   /// turns on signposts in Instruments.
   @objc public var osLogEnabled: Bool = false
 
+  /// Whether to set `log_user_id` on logged events (on
+  /// outer `LogRequest`).
+  @objc public var metricsLoggerIncludesLogUserIDs: Bool = true
```

Comment:
Why give any control to the client on whether the fields are set?  Why not always set?  Is it because, depending on the client, the values might be wrong?

Reply:
is there a way to only populate parent ids that the sdk is logging/managing itself without resorting these config settings?  for hipcamp, only impressions are meant to be logged by the client sdk, so the should really only set the impression id.

Reply:
We need these settings because the desired behavior is different for Snackpass and Hipcamp. For Snackpass, the initial RPC needs these IDs to be present before any user/session/view has been logged. For Hipcamp, we want the opposite.

I investigated Jin's suggestion, and it's complicated to get the code to behave in as desired with both Snackpass and Hipcamp in mind. It's certainly possible, but also grows increasingly complex with more clients in the future.

Given these different requirements for these two clients, I'd rather have explicit configuration than complicated logic in the client to control this behavior. We can revisit this later, with more clients using this library.

Reply:
Gotcha.  Yea, I agree.  We can re-eval later.

Reply:
Actually, here's a version of this fix that omits parent IDs from events unless those parent events have been logged: https://github.com/promotedai/ios-metrics-sdk/pull/120/files

If this change is generalizable to future clients, we could go that way instead, and avoid extra configuration.